packages/react/src/editor/BlockNoteTheme.ts

```diff
@@ -32,133 +32,165 @@ export type Theme = {
   fontFamily: string;
 };
 
-export const applyCSSVariablesFromTheme = (
+const camelCaseToKebabCase = (str: string) =>
+  str.replace(/([a-z])([A-Z])/g, "$1-$2").toLowerCase();
+
+const cssVariablesHelper = (
   theme: Theme,
-  editorDOM: HTMLElement
+  editorDOM: HTMLElement,
+  unset = false
 ) => {
-  editorDOM.style.setProperty(
-    "--bn-color-editor-text",
-    theme.colors.editor.text
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-editor-background",
-    theme.colors.editor.background
-  );
-  editorDOM.style.setProperty("--bn-color-menu-text", theme.colors.menu.text);
-  editorDOM.style.setProperty(
-    "--bn-color-menu-background",
-    theme.colors.menu.background
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-tooltip-text",
-    theme.colors.tooltip.text
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-tooltip-background",
-    theme.colors.tooltip.background
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-hovered-text",
-    theme.colors.hovered.text
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-hovered-background",
-    theme.colors.hovered.background
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-selected-text",
-    theme.colors.selected.text
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-selected-background",
-    theme.colors.selected.background
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-disabled-text",
-    theme.colors.disabled.text
-  );
-  editorDOM.style.setProperty(
-    "--bn-color-disabled-background",
-    theme.colors.disabled.background
-  );
+  // Colors
+  for (const [colorSchemeKey, colorSchemeValue] of Object.entries(
```

Comment:
why are colors a separate case? Why can't we make this code completely generic where `{ a: { b: {c: 4 }}}` is translated to `setProperty("--bn-a-b-c", 4);`? Maybe I'm missing something?